casr/src/util.rs

```diff
@@ -293,3 +284,532 @@ pub fn get_output(command: &mut Command, timeout: u64, error_on_timeout: bool) -
             .with_context(|| format!("Couldn't launch {command:?}"))
     }
 }
+
+/// Create output, timeout and oom directories
+///
+/// # Arguments
+///
+/// `matches` - tool argumnets
+///
+/// Return value
+///
+/// Paths to (output, oom, timeout) directories
+fn initialize_dirs(matches: &clap::ArgMatches) -> Result<(&PathBuf, PathBuf, PathBuf)> {
+    // Get output dir
+    let output_dir = matches.get_one::<PathBuf>("output").unwrap();
+    if !output_dir.exists() {
+        fs::create_dir_all(output_dir).with_context(|| {
+            format!("Couldn't create output directory {}", output_dir.display())
+        })?;
+    } else if output_dir.read_dir()?.next().is_some() {
+        bail!("Output directory is not empty.");
+    }
+    // Get oom dir
+    let oom_dir = output_dir.join("oom");
+    if fs::create_dir_all(&oom_dir).is_err() {
+        bail!("Failed to create dir {}", &oom_dir.to_str().unwrap());
+    }
+    // Get timeout dir
+    let timeout_dir = output_dir.join("timeout");
+    if fs::create_dir_all(&timeout_dir).is_err() {
+        bail!("Failed to create dir {}", &timeout_dir.to_str().unwrap());
+    }
+
+    Ok((output_dir, oom_dir, timeout_dir))
+}
+
+#[derive(Debug, Clone, Default)]
+/// Information about crash to reproduce it.
+pub struct CrashInfo {
```

Comment:
I suppose we need more modules than just `util.rs`. Or we can even move some code to libCASR.

Reply:
done